### PR TITLE
GeometryFileSink

### PR DESCRIPTION
**Pull request by caspervdw**
This adds a block that writes geometries to a file. The formats geojson, shapefile, geopackage and GML are supported.

The filesink saves files in a directory. This is optimized for large exports (where tiling is necessary). Because most files are not 'appendable', saving different files was the best option. Also, it prevents 'multiple write' events.

At the end you have a folder of files that can be merged into one using `GeometryFileSink.merge_files`. That also cleans up the source files (if requested).

A common usage pattern would be like this:
```
source = GeometryFileSource(...)
sink = GeometryFileSink(source, tmp_local_path, "geojson")
tiler = GeometryTiler(sink, size=10000, projection="EPSG:28992)
tiler.get_data(**some_large_request)
GeometryFileSink.merge_files(tmp_local_path, target, remove_sources=True)
```

To make this easier, this API is also there (this uses a tmpdir):
```
source.to_file(target, tile_size=10000, **some_large_request)
```

### CODE REVIEW

**Comment by caspervdw** [comment]
@arjanverkerk  Klaar!

File: dask_geomodeling/tests/test_geometry_sinks.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            assert len(df) == 1
+            assert df.crs["init"] == "epsg:3857"
+
+    def test_to_file(self):
+        self.source.to_file(self.path + ".geojson", **self.request)
+        actual = gpd.read_file(self.path + ".geojson")
+        # compare dataframes without checking the order of records / columns
+        assert_frame_equal(actual, self.expected, check_like=True)
+
+    def test_to_file_with_tiling(self):
```
Comment (arjanverkerk) [inline dask_geomodeling/tests/test_geometry_sinks.py]: This does not tile?

File: dask_geomodeling/geometry/sinks.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+    @staticmethod
+    def process(data, process_kwargs):
+        if "features" not in data or len(data["features"]) == 0:
+            return data  # do nothing for non-feature or empty requests
+
+        features = data["features"].copy()
+        projection = data["projection"]
+        crs = utils.get_crs(projection)
+        path = utils.safe_abspath(process_kwargs["url"])
```
[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/geometry/sinks.py]: We serialize a user-supplied path to an URL and then transform it back to path again here. The 'safe' part is actually a bit misleading, it doesn't check anything anymore. Previously it checked if the path was inside the "ROOT" setting, but not anymore.

File: dask_geomodeling/geometry/sinks.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+            features[index_name] = features.index
+
+        # copy the dataframe and rename the columns
+        features = features[["geometry"] + list(fields.values())]
+
+        # rename the columns
+        features.columns = ["geometry"] + list(fields.keys())
+
+        # generate the file
+        features.crs = crs  # be sure about the CRS
```
[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/geometry/sinks.py]: We *should* be able to safely omit it. But we have been sloppy specifying the `crs` attribute and indeed there are some mistakes here and there. I'll try to address these in this PR.

[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/geometry/sinks.py]: OK I fixed the GeometryFileSource and also added a test at https://github.com/nens/django-geoblocks/pull/37 to confirm that it works for the GeoDjangoSource already.

**Pull request merged**